setup.py

```diff
@@ -0,0 +1,57 @@
+# ============================================================================ #
+# Copyright (c) 2022 - 2023 NVIDIA Corporation & Affiliates.                   #
+# All rights reserved.                                                         #
+#                                                                              #
+# This source code and the accompanying materials are made available under     #
+# the terms of the Apache License 2.0 which accompanies this distribution.     #
+# ============================================================================ #
+import os
+import sys
+try:
+    import skbuild
+except ImportError:
+    print("Trying to install required module: skbuild")
+    os.system(f"{sys.executable} -m pip install scikit-build")
+import skbuild
+import setuptools
+
+__version__ = "0.0.3"
+
+# The setup script gets called twice when installing from source
+# with `pip install .` . Once for the `egg_info` subcommand and another
+# for `install`. We will only install the missing dependencies once, for
+# the `egg_info` subcommand (arbitrary choice).
+if (sys.argv[1] == 'egg_info'):
+    script_path = os.getcwd() + "/scripts/install_wheel_dependencies.sh"
+    os.system(f"bash {script_path}")
+
+# FIXME: Linux machines default to dist-packages unless the `--user` flag is provided to
```

Comment:
Please add a link to a GitHub issue for any remaining Fixme's and Todo's in the code that is checked in.